networkx/generators/classic.py

```diff
@@ -322,7 +322,7 @@ def grid_2d_graph(m,n,periodic=False,create_using=None):
         boundary nodes via periodic boundary conditions.
     """
     G=empty_graph(0,create_using)
-    G.name="grid_2d_graph"
+    G.name="grid_2d_graph(%d,%d)"%(m,n)
```

Comment:
Yes, that is the new Python way.
No need to update the whole file in this PR.  If you are inspired do that in a separate issue.